src/bidiMapper/domains/network/NetworkManager.ts

```diff
@@ -87,13 +87,21 @@ export class NetworkManager {
       networkStorage
     );
 
+    const detachedListener = (
+      params: Protocol.Target.DetachedFromTargetEvent
+    ) => {
+      if (cdpTarget.cdpClient.sessionId === params.sessionId) {
+        networkManager.#networkStorage.disposeRequestMap();
+        cdpTarget.browserCdpClient.off(
+          'Target.detachedFromTarget',
+          detachedListener
+        );
+      }
+    };
+
     cdpTarget.browserCdpClient.on(
       'Target.detachedFromTarget',
-      (params: Protocol.Target.DetachedFromTargetEvent) => {
-        if (cdpTarget.cdpClient.sessionId === params.sessionId) {
-          networkManager.#networkStorage.disposeRequestMap();
-        }
-      }
+      detachedListener
```

Comment:
Is the issue hypothetical or real? I don't think there is a difference between the before and the after version? The listener still retains the target and the network manager.

Reply:
Hypothetical, but from my understanding `detachedListener` loses scope after the return from `create` and the only reference lives inside the `EventEmitter`. Also we explicitly call the `off` method on the `EventEmitter` now.
Graph of ref should be `EventEmitter` -> `detachedListener` -> `networkManager`, so calling `off` should break the first chain.

Reply:
It also looses scope when it is inline, calling the `off` should release it just as well.

Reply:
How do we get the reference to call `off` for your case?